keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -547,57 +497,68 @@ This includes 3 scenarios:
 For each exported service, one `ServiceExport` will exist in each cluster that
 runs the service. The mcsd-controller will create and maintain a derived
 `ServiceImport` in each cluster within the supercluster (see: [constraints and
-conflict resolution](#constraints-and-conflict-resolution)). If all `ServiceExport` instances are deleted, each
-`ServiceImport` will also be deleted from all clusters.
+conflict resolution](#constraints-and-conflict-resolution)). If all
+`ServiceExport` instances are deleted, each `ServiceImport` will also be deleted
+from all clusters.
 
 Since a given `ServiceImport` may be backed by multiple `EndpointSlices`, a
 given `EndpointSlice` will reference its `ServiceImport` using the label
 `multicluster.kubernetes.io/imported-service-name` similarly to how an
 `EndpointSlice` is associated with its `Service` in a single cluster. Each
 imported `EndpointSlice` will also have a
-`multicluster.kubernetes.io/source-cluster` label with a registry-scoped unique
-identifier for the cluster.
+`multicluster.kubernetes.io/source-cluster` label with the cluster name, a
+registry-scoped unique identifier for the cluster.
 
 ```golang
-// ServiceImport declares that the specified service should be exported to other clusters.
+// ServiceImport describes a service imported from clusters in a supercluster.
 type ServiceImport struct {
- metav1.TypeMeta `json:",inline"`
- // +optional
- metav1.ObjectMeta `json:"metadata,omitempty"`
- // +optional
- Spec ServiceImportSpec `json:"spec,omitempty"`
+  metav1.TypeMeta `json:",inline"`
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+  // +optional
+  Spec ServiceImportSpec `json:"spec,omitempty"`
+  // +optional
+  Status ServiceImportStatus `json:"status,omitempty"`
 }
 
-// ServiceImportSpec contains the current status of an imported service and the
-// information necessary to consume it
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+  // SuperclusterIP are only accessible via the Supercluster IP.
+  SuperclusterIP ServiceImportType = "SuperclusterIP"
+  // Headless services allow backend pods to be addressed directly.
+  Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
 type ServiceImportSpec struct {
   // +patchStrategy=merge
   // +patchMergeKey=port
   // +listType=map
   // +listMapKey=port
   // +listMapKey=protocol
   Ports []ServicePort `json:"ports"`
+  // +optional
+  IP string `json:"ip,omitempty"`
```

Comment:
Is it allowed for a user to create a ServiceImport?  If so, does this trigger kube-proxy to allocate that IP?  What is it checked against at creation time?

Reply:
A user would be able to create a ServiceImport, and creating it would cause kube-proxy to allocate that IP. This is basically the bash-script-controller version of MCS. The ability for a user to claim an arbitrary IP within their cluster is a concern, it would be great to have a better model for IP allocation

Reply:
Warning added